substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -119,84 +419,1339 @@ protected int peekData(DatagramPacket p) throws IOException {
+            //  1874                          "Socket closed");
+            throw new SocketException("Socket closed");
+            //  1875          return;
+        } else {
+            //  1877          fd = (*env)->GetIntField(env, fdObj, IO_fd_fdID);
+            fd = PosixUtils.getFD(fdObj);
+        }
+        //  1879      /* setsockopt to be correct TTL */
+        //  1880  #ifdef AF_INET6
+        if (IsDefined.socket_AF_INET6()) {
+            //  1881  #ifdef __linux__
+            if (IsDefined.__linux__()) {
+                //  1882      setTTL(env, fd, ttl);
+                Util_java_net_PlainDatagramSocketImpl.setTTL(fd, ttl);
+                //  1883      if (ipv6_available()) {
+                if (JavaNetNetUtil.ipv6_available()) {
+                    //  1884          setHopLimit(env, fd, ttl);
+                    Util_java_net_PlainDatagramSocketImpl.setHopLimit(fd, ttl);
+                }
+            } else {
+                //  1886  #else  /*  __linux__ not defined */
+                //  1887      if (ipv6_available()) {
+                if (JavaNetNetUtil.ipv6_available()) {
+                    //  1888          setHopLimit(env, fd, ttl);
+                    Util_java_net_PlainDatagramSocketImpl.setHopLimit(fd, ttl);
+                } else {
+                    //  1890          setTTL(env, fd, ttl);
+                    Util_java_net_PlainDatagramSocketImpl.setTTL(fd, ttl);
+                }
+            }
+            //  1892  #endif  /* __linux__ */
+            //  1893  #else
+        } else {
+            //  1894      setTTL(env, fd, ttl);
+            Util_java_net_PlainDatagramSocketImpl.setTTL(fd, ttl);
+        }
+        //  1895  #endif  /* AF_INET6 */
     }
+    // @formatter:on
+
+
 
+    //  1909  /*
+    //  1910   * Class:     java_net_PlainDatagramSocketImpl
+    //  1911   * Method:    getTTL
+    //  1912   * Signature: ()B
+    //  1913   */
+    //  1914  JNIEXPORT jint JNICALL
+    //  1915  Java_java_net_PlainDatagramSocketImpl_getTimeToLive(JNIEnv *env, jobject this) {
     @Substitute
     @SuppressWarnings({"static-method"})
-    protected int getTimeToLive() {
-        throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.getTimeToLive()");
+    protected int getTimeToLive() throws IOException {
+        //  1917      jobject fdObj = (*env)->GetObjectField(env, this, pdsi_fdID);
+        FileDescriptor fdObj = Util_java_net_DatagramSocketImpl.as_Target_java_net_DatagramSocketImpl(this).fd;
+
+        //  1918      jint fd = -1;
+        int fd = -1;
+
+        //  1920      if (IS_NULL(fdObj)) {
+        if (fdObj == null) {
+            //  1921          JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException",
+            //  1922                          "Socket closed");
+            //  1923          return -1;
+            //throw new SocketException("Socket closed");
+            return -1;
```

Comment:
This is a weird one, because you have to choose between throwing the exception or returning a value.  Java code that calls this through JNI will get the exception before (instead of) the return value.  I think you should throw the exception.  Unless you have some local Java code that wants the return value and will do the right thing with it.